Generated validator range attributes use float literals for integer fields

`oas3-gen v0.26.1` emits float literals in `#[validate(range(...))]` on `i64` fields, which does not compile.

Given `"type": "integer", "minimum": -1, "maximum": 1` or `"exclusiveMinimum": 0`, the generated output is:

```rust
#[validate(range(min = -1.0, max = 1.0))]
pub field: i64,

#[validate(range(exclusive_min = 0.0))]
pub other_field: i64,
```

These should be integer literals (`-1`, `1`, `0`), not floats:

```
error[E0308]: mismatched types
    |     #[validate(range(min = -1.0, max = 1.0))]
    |                            ^^^^ expected `i64`, found floating-point number
```

--- a/crates/oas3-gen/src/generator/ast/mod.rs
+++ b/crates/oas3-gen/src/generator/ast/mod.rs
@@ -37,7 +37,7 @@ pub use status_codes::StatusCodeToken;
 pub use tokens::{
   DefaultAtom, EnumToken, EnumVariantToken, FieldNameToken, MethodNameToken, StructToken, TraitToken, TypeAliasToken,
 };
-pub use types::{RustPrimitive, TypeRef};
+pub use types::{Rounding, RustPrimitive, TypeRef};
 pub use validation_attrs::{RegexKey, ValidationAttribute};
 
 pub use crate::generator::ast::fields::{FieldCollection, FieldDef};

--- a/crates/oas3-gen/src/generator/ast/tests/validation_attrs.rs
+++ b/crates/oas3-gen/src/generator/ast/tests/validation_attrs.rs
@@ -66,6 +66,42 @@ fn test_validation_attribute_range_display() {
     attr_float.to_token_stream().to_string(),
     "range (min = 0.5 , max = 1.0)"
   );
+
+  let attr_float_encoded_int = ValidationAttribute::Range {
+    primitive: RustPrimitive::I64,
+    min: Some(serde_json::json!(-1.0).as_number().unwrap().clone()),
+    max: Some(serde_json::json!(1.0).as_number().unwrap().clone()),
+    exclusive_min: None,
+    exclusive_max: None,
+  };
+  assert_eq!(
+    attr_float_encoded_int.to_token_stream().to_string(),
+    "range (min = - 1i64 , max = 1i64)"
+  );
+
+  let attr_fractional = ValidationAttribute::Range {
+    primitive: RustPrimitive::I64,
+    min: Some(serde_json::json!(1.5).as_number().unwrap().clone()),
+    max: Some(serde_json::json!(2.5).as_number().unwrap().clone()),
+    exclusive_min: None,
+    exclusive_max: None,
+  };
+  assert_eq!(
+    attr_fractional.to_token_stream().to_string(),
+    "range (min = 2i64 , max = 2i64)"
+  );
+
+  let attr_fractional_exclusive = ValidationAttribute::Range {
+    primitive: RustPrimitive::I64,
+    min: None,
+    max: None,
+    exclusive_min: Some(serde_json::json!(1.5).as_number().unwrap().clone()),
+    exclusive_max: Some(serde_json::json!(2.5).as_number().unwrap().clone()),
+  };
+  assert_eq!(
+    attr_fractional_exclusive.to_token_stream().to_string(),
+    "range (exclusive_min = 1i64 , exclusive_max = 3i64)"
+  );
 }
 
 #[test]

--- a/crates/oas3-gen/src/generator/ast/types.rs
+++ b/crates/oas3-gen/src/generator/ast/types.rs
@@ -187,6 +187,13 @@ impl From<&serde_json::Value> for TypeRef {
   }
 }
 
+/// Direction for rounding a float-encoded bound to an integer literal
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Rounding {
+  Ceil,
+  Floor,
+}
+
 /// Rust primitive and standard library types
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Default, strum::Display)]
 #[strum(serialize_all = "lowercase")]
@@ -323,6 +330,29 @@ impl RustPrimitive {
     } else {
       num.to_string()
     }
+  }
+
+  pub fn format_range_bound(&self, num: &Number, rounding: Rounding) -> String {
+    if self.is_float() {
+      return self.format_number(num);
+    }
+    if let Some(value) = num.as_i64() {
+      return render_integer(self, value);
+    }
+    if let Some(value) = num.as_u64() {
+      return render_unsigned_integer(self, value);
+    }
+    if let Some(f) = num.as_f64() {
+      let rounded = match rounding {
+        Rounding::Ceil => f.ceil(),
+        Rounding::Floor => f.floor(),
+      };
+      #[allow(clippy::cast_possible_truncation)]
+      if rounded >= i64::MIN as f64 && rounded <= i64::MAX as f64 {
+        return render_integer(self, rounded as i64);
+      }
+    }
+    num.to_string()
   }
 
   fn format_string_value(&self, s: &str) -> String {

--- a/crates/oas3-gen/src/generator/ast/validation_attrs.rs
+++ b/crates/oas3-gen/src/generator/ast/validation_attrs.rs
@@ -3,7 +3,7 @@ use proc_macro2::TokenStream;
 use quote::{ToTokens, quote};
 use serde_json::Number;
 
-use crate::generator::ast::{RustPrimitive, StructToken, TypeRef, types::render_unsigned_integer};
+use crate::generator::ast::{Rounding, RustPrimitive, StructToken, TypeRef, types::render_unsigned_integer};
 
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct RegexKey {
@@ -185,19 +185,31 @@ impl ToTokens for ValidationAttribute {
       } => {
         let mut parts = vec![];
         if let Some(m) = min {
-          let lit = primitive.format_number(m).parse::<TokenStream>().unwrap();
+          let lit = primitive
+            .format_range_bound(m, Rounding::Ceil)
+            .parse::<TokenStream>()
+            .unwrap();
           parts.push(quote! { min = #lit });
         }
         if let Some(m) = max {
-          let lit = primitive.format_number(m).parse::<TokenStream>().unwrap();
+          let lit = primitive
+            .format_range_bound(m, Rounding::Floor)
+            .parse::<TokenStream>()
+            .unwrap();
           parts.push(quote! { max = #lit });
         }
         if let Some(m) = exclusive_min {
-          let lit = primitive.format_number(m).parse::<TokenStream>().unwrap();
+          let lit = primitive
+            .format_range_bound(m, Rounding::Floor)
+            .parse::<TokenStream>()
+            .unwrap();
           parts.push(quote! { exclusive_min = #lit });
         }
         if let Some(m) = exclusive_max {
-          let lit = primitive.format_number(m).parse::<TokenStream>().unwrap();
+          let lit = primitive
+            .format_range_bound(m, Rounding::Ceil)
+            .parse::<TokenStream>()
+            .unwrap();
           parts.push(quote! { exclusive_max = #lit });
         }
         quote! { range(#(#parts),*) }

